src/Bridges/quadtosocbridge.jl

```diff
@@ -0,0 +1,197 @@
+model by checking the existence of bounds on variables as shown below.
+
+### Second-order cone
+
+If `Q` is diagonal and has eigenvalues `(1, 1, -1)`, the inequality
+``x^2 + x^2 \\le z^2`` combined with ``z \\ge 0`` defines the Lorenz cone (i.e.
+the second-order cone) but when combined with ``z \\le 0``, it gives the
+opposite of the second order cone. Therefore, we need to check if the variable
+`z` has a lower bound 0 or an upper bound 0 in order to determine which cone is
+
+### Rotated second-order cone
+
+The matrix `Q` corresponding to the inequality ``x^2 \\le 2yz`` has one
+eigenvalue 1 with eigenvectors `(1, 0, 0)` and `(0, 1, -1)` and one eigenvalue
+`-1` corresponding to the eigenvector `(0, 1, 1)`. Hence if we intersect this
+union of two convex cone with the halfspace ``x + y \\ge 0``, we get the rotated
+second-order cone and if we intersect it with the halfspace ``x + y \\le 0`` we
+get the opposite of the rotated second-order cone. Note that `y` and `z` have
+the same sign since `yz` is nonnegative hence ``x + y \\ge 0`` is equivalent to
+``x \\ge 0`` and ``y \\ge 0``.
+
+### Note
+
+The check for existence of bound can be implemented (but inefficiently) with the
+current interface but if bound is removed or transformed (e.g. `≤ 0` transformed
+into `≥ 0`) then the bridge is no longer valid. For this reason the homogeneous
+version of the bridge is not implemented yet.
+"""
+struct QuadtoSOCBridge{T} <: AbstractBridge
+    soc::CI{MOI.VectorAffineFunction{T}, MOI.RotatedSecondOrderCone}
+    dimension::Int  # dimension of the SOC constraint
+    less_than::Bool # whether the constraint was ≤ or ≥
+    set_constant::T # the constant that was on the set
+end
+function QuadtoSOCBridge{T}(model, func::MOI.ScalarQuadraticFunction{T},
+                            set::Union{MOI.LessThan{T},
+                                       MOI.GreaterThan{T}}) where T
+    set_constant = MOIU.getconstant(set)
+    less_than = set isa MOI.LessThan
+    if !less_than
+        set_constant = -set_constant
+    end
+    Q, index_to_variable_map = matrix_from_quadratic_terms(func.quadratic_terms)
+    if !less_than
+        Compat.rmul!(Q, -1)
+    end
+    # We have L × L' ≈ Q[p, p]
+    L, p = try
+        @static if VERSION >= v"0.7-"
+            F = cholesky(Symmetric(Q))
+            sparse(F.L), F.p
+        else
+            F = cholfact(Symmetric(Q))
+            sparse(F[:L]), F[:p]
+        end
+    catch err
+        if err isa PosDefException
+            error("The optimizer supports second-order cone constraints and",
+                  " not quadratic constraints but you entered a quadratic",
+                  " constraint of type: `$(typeof(func))`-in-`$(typeof(set))`.",
+                  " A bridge attempted to transform the quadratic constraint",
+                  " to a second order cone constraint but the constraint is not",
+                  " convex.")
```

Comment:
This is incorrect. `cholfact` will fail when `Q` is only positive semi-definite, but that's a perfectly valid convex constraint.

Reply:
You could try LDLT instead right? Or bunch-kaufman if dense